R/eval.R

```diff
@@ -69,10 +69,9 @@ evaluate <- function(input,
   }
 
   if (new_device) {
-    # Start new graphics device and clean up afterwards
-    if (identical(grDevices::pdf, getOption("device"))) {
-      dev.new(file = NULL)
-    } else dev.new()
+    # Ensure we have a graphics device available for recording, but choose
+    # one that's available on all platforms and doesn't write to disk
+    pdf(file = NULL)
```

Comment:
We could alternatively use something like `devoid::void_dev()` but that would add a dependency, and there doesn't appear to be any performance advantage:

```R
x <- rnorm(1e5)
y <- rnorm(1e5)

f <- function(dev) {
  dev()
  plot(x, y)
  dev.off()
}

bench::mark(
  f(devoid::void_dev),
  f(\() pdf(NULL))
)
#> # A tibble: 2 × 3
#> expression                   min   median
#> <bch:expr>              <bch:tm> <bch:tm>
#> 1 f(devoid::void_dev)       10.8ms   11.1ms
#> 2 f(function() pdf(NULL))   10.9ms   11.2ms
```